gn3/computations/datasets.py

```diff
@@ -37,21 +35,22 @@ def retrieve_trait_sample_data(dataset,
 
     if dataset_query:
         formatted_query = dataset_query % sample_query_values[dataset_type]
-        results = fetch_from_db_sample_data(formatted_query, mock.Mock())
+
+        results = fetch_from_db_sample_data(formatted_query, database)
 
     return results
 
 
 def fetch_from_db_sample_data(formatted_query: str, database_instance) -> List:
     """this is the function that does the actual fetching of\
     results from the database"""
-    cursor = database_instance.cursor()
-    _conn = database_connector
-    # conn, cursor = database_connector()
-    # cursor = conn.cursor()
+    try:
+        cursor = database_instance.cursor()
+        cursor.execute(formatted_query)
+        results = cursor.fetchall()
 
-    cursor.execute(formatted_query)
-    results = cursor.fetchall()
+    except Exception as error:
+        raise error
```

Comment:
**issue:** Wrap connection ops in with syntax, since it auto-magically closes the connection for you; instead of leaving that to python/ flask.